.azure/pipelines/jobs/default-build.yml

```diff
@@ -97,12 +94,13 @@ jobs:
     BuildScriptArgs: ${{ parameters.buildArgs }}
     BuildConfiguration: ${{ parameters.configuration }}
     BuildDirectory: ${{ parameters.buildDirectory }}
-    VSTS_OVERWRITE_TEMP: false # Workaround for https://github.com/dotnet/core-eng/issues/2812
-    ${{ if eq(parameters.codeSign, 'true') }}:
+    ${{ if eq(parameters.agentOs, 'Windows') }}:
+      JAVA_HOME: $(Agent.WorkFolder)/.tools/jdk
```

Comment:
I think this may be the problem. According to my interpretation of https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables. this folder is not guaranteed to be cleaned up.

Reply:
Also should you be using back slashes instead of forward slashes as it's a file path not URL?  So that `/.tools/jdk` becomes `\.tools\jdk`

Reply:
Updated.